spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcMessageStore.java

```diff
@@ -204,20 +200,31 @@ public String getSql() {
 
 	/**
 	 * Convenient constructor for configuration use.
+	 * @deprecated since 5.0 in favor of {@link #JdbcMessageStore(DataSource)}
 	 */
+	@Deprecated
 	public JdbcMessageStore() {
-		this.deserializer = new DeserializingConverter();
-		this.serializer = new SerializingConverter();
+		throw new UnsupportedOperationException("Please, use DataSource-based constructor");
 	}
```

Comment:
Usually deprecated methods still work, we just want to discourage use; since this is a major release, and given this exception, I see no point in deprecating - just get rid of it.

We could/should deprecate in 4.3, though.